src/pybind/ceph_argparse.py

```diff
@@ -362,7 +371,98 @@ def argdesc(self, attrs):
         if self.goodchars:
             attrs['goodchars'] = self.goodchars
         return super().argdesc(attrs)
+    
+
+class CephArgtypeConverter(CephArgtype):
+    def convert(self, value):  # noqa # pylint: disable=unused-variable
+        raise NotImplementedError()
+
+
+class CLI_ANNOTATIONS:
+    MemorySize = 'cli_memory_size'
+
+
+def _get_annotation(tp):
+    if get_origin(tp) is Annotated:
+        annotated_args = get_args(tp)
+        if len(annotated_args) >= 2:
+            metadata_type = annotated_args[1]
+            return metadata_type
+
+        
+def _get_annotated_type_converter(annotation) -> Optional[CephArgtypeConverter]:
+    if annotation == CLI_ANNOTATIONS.MemorySize:
+        return CephSizeBytes()
+    
+B = "B"
+K, KB, KIB = "K", "KB", "KiB"
+M, MB, MIB = "M", "MB", "MiB"
+G, GB, GIB = "G", "GB", "GiB"
+T, TB, TIB = "T", "TB", "TiB"
+P, PB, PIB = "P", "PB", "PiB"
+MULTIPLES = ['', K, M, G, T, P]
+UNITS = {
+    f"{prefix}{suffix}": 1024 ** mult
+    for mult, prefix in enumerate(MULTIPLES)
+    for suffix in ['', 'B', 'iB']
+    if not (prefix == '' and suffix == 'iB')
+}
```

Comment:
BTW, I'd put these under the CephSizeBytes class. IMHO data structures should be in the scope where they're used (that's basically encapsulation).